lib/matplotlib/tests/test_colorbar.py

```diff
@@ -298,6 +298,26 @@ def test_colorbar_ticks():
     assert len(cbar.ax.xaxis.get_ticklocs()) == len(clevs)
 
 
+@cleanup
+def test_colorbar_get_ticks():
+    # test feature for #5792
+    plt.figure()
+    data = np.arange(1200).reshape(30, 40)
+
+    plt.subplot()
+    plt.contourf(data)
+
+    userTicks = plt.colorbar(use_gridspec=True, ticks=[0, 600, 1200])
+    assert userTicks.get_ticks().tolist() == [0, 600, 1200]
+
+    defTicks = plt.colorbar(use_gridspec=True, orientation='horizontal')
+    assert defTicks.get_ticks().tolist() == [0.0, 150.0, 300.0, 450.0, 600.0,
```

Comment:
If the default for selecting the contour levels were to change, this would break.  I suggest generating an array of levels, using that as the `levels` kwarg to `contourf` and then testing against that same array.